dojo/tools/intsights/parser.py

```diff
@@ -0,0 +1,172 @@
+        required_keys = ['alert_id', 'title', 'severity', 'status']
+
+        alerts = []
+        invalid_alerts = []
+
+        content = csv_file.read()
+        if type(content) is bytes:
+            content = content.decode('utf-8')
+        csv_reader = csv.DictReader(io.StringIO(content), delimiter=',', quotechar='"')
+
+        # Don't bother parsing if the keys don't match exactly what's expected
+        if collections.Counter(default_keys) == collections.Counter(csv_reader.fieldnames):
+            for alert in csv_reader:
+                alert['alert_id'] = alert.pop('Alert ID')
+                alert['title'] = alert.pop('Title')
+                alert['description'] = alert.pop('Description')
+                alert['severity'] = alert.pop('Severity')
+                alert['type'] = alert.pop('Type')
+                alert['source_date'] = alert.pop('Source Date (UTC)')
+                alert['report_date'] = alert.pop('Report Date (UTC)')
+                alert['network_type'] = alert.pop('Network Type')
+                alert['source_url'] = alert.pop('Source URL')
+                alert['assets'] = alert.pop('Assets')
+                alert['tags'] = alert.pop('Tags')
+                alert['status'] = alert.pop('Status')
+                alert['alert_link'] = alert.pop('Alert Link')
+                alert.pop('Assignees')
+                alert.pop('Remediation')
+                alert.pop('Closed Reason')
+                alert.pop('Rating')
+                for key in required_keys:
+                    if not alert[key]:
+                        invalid_alerts.append(alert)
+
+                if alert not in invalid_alerts:
+                    alerts.append(alert)
+        else:
+            self._LOGGER.error('The CSV file has one or more missing or unexpected header values')
+
+        return alerts
+
+    def _build_finding_description(self, alert: dict) -> str:
+        """
+        Builds an IntSights FInding description from various pieces of information.
+        Args:
+            alert: The parsed alert dictionary
+        Returns: A markdown formatted description
+        """
+
+        description = f'{alert["description"]}' \
+                      f'\n\n----' \
+                      f'\r\n**Date Found**: {alert.get("report_date", "None provided")}' \
+                      f'\n\n----' \
+                      f'\r\n**Type**: {alert.get("type", "None provided")}' \
+                      f'\r\n**Source**: {alert.get("source_url", "None provided")}' \
+                      f'\r\n**Source Date**: {alert.get("source_date", "None provided")}' \
+                      f'\r\n**Source Network Type**: {alert.get("network_type", "None provided")}' \
+                      f'\n\n----' \
+                      f'\r\n**Assets Affected**: {alert.get("assets", "None provided")}' \
+                      f'\n\n----' \
+                      f'\r\n**Alert Link**: {alert.get("alert_link", "None provided")}'
```

Comment:
1) for this kind of structure, you can use something like:
```Python
"\n".join([
  part1,
  part2,
])
```
2) as it's Markdown syntax, you can add ` chars like that:

```Python
                description = "\n".join([
                    "**IdRef:** `" + rule_result.attrib['idref'] + "`",
                    "**Title:** `" + title + "`",
                ])
```
see https://github.com/DefectDojo/django-DefectDojo/blob/4a9a4e46525874e89daa2d677679bbbe1a89494c/dojo/tools/openscap/parser.py#L54-L57

Reply:
it produces better formatting when rendering the HTML
![image](https://user-images.githubusercontent.com/1694940/116091807-7eacca80-a6a5-11eb-9523-2b9ffb8f9973.png)